app/shared/api-services/poll/answer-factory.js

```diff
@@ -27,25 +26,20 @@ angular.module('apiServices.answer.factory', [
       return $http.put(answerUrl + newAnswer.id, {
         answer_text: newAnswer.answer_text
       }).then(function(response) {
-        if(response.status === 200) {
-          return response.data;
-        } else {
-          return $q.reject(newAnswer.answer_text + ' update unsuccessful');
-        }
+        return response.data;
       });
     }
   }
 
   function postAnswer(answer, pollId) {
+    if(typeof pollId !== 'number' || !answer.answer_text) {
```

Comment:
We're all over the board on this one, so I hate to mention this so late, but since we have a lodash dep in this factory specifically we might as well use the lodash `_.is*` methods for checks like this.
```js
_.isNumber(pollId) || _.isUndefined(answer.answer_text)
```
or somesuch thing.